trie/iterator.go

```diff
@@ -17,25 +17,72 @@
 package trie
 
 import (
-	log "github.com/inconshreveable/log15"
+	"fmt"
+	//log "github.com/inconshreveable/log15"
 )
 
 // Print prints the trie through pre-order traversal
 func (t *Trie) Print() {
-	log.Info("printing trie...")
-	t.print(t.root, nil)
+	fmt.Println("printing trie...")
+	t.print(t.root, nil, false)
 }
 
-func (t *Trie) print(current node, prefix []byte) {
+func (t *Trie) PrintEncoding() {
+	t.print(t.root, nil, true)
+}
+
+func (t *Trie) print(current node, prefix []byte, withEncoding bool) {
+	h, err := NewHasher()
+	if err != nil {
+		fmt.Printf("new hasher err %s\n", err)
+	}
+	var encoding []byte
+	var hash []byte
+	if withEncoding && current != nil {
+		encoding, err = current.Encode()
+		if err != nil {
+			fmt.Printf("encoding err %s\n", err)
+		}
+		hash, err = h.Hash(current)
+		if err != nil {
+			fmt.Printf("hashing err %s\n", err)
+		}
+	}
+
 	switch c := current.(type) {
 	case *branch:
-		log.Info("branch key %x children %b value %s\n", nibblesToKeyLE(append(prefix, c.key...)), c.childrenBitmap(), c.value)
+		fmt.Printf("branch key %x children %b value %x\n", nibblesToKeyLE(append(prefix, c.key...)), c.childrenBitmap(), c.value)
+		if withEncoding {
+			fmt.Printf("branch encoding ")
```

Comment:
is there a particular reason we are printing messages to the terminal this often? Makes for messy logs

Reply:
this function is for printing/debugging, i can try to change it to log15

Reply:
also idk how to throw the `printHexBytes` function into log15, can we just leave it for now? i feel like fmt looks more readable for trie debugging than a logger

Reply:
Okay but make a task in zube we need to be consistent throughout each package how we are logging and setting up debugging logs. We will have to go back and change the logs to reflect the rest of the package files.

Reply:
ok done. David pointed out this should probably be made into an actual iterator so I'm going to work on that